Join Blend as core node (#340)
status-im/ift-docs: docs/blockchain/blend/join-the-blend-network-as-a-core-node.md

---
title: Join the Blend network as a core node
doc_type: procedure
product: blockchain
topics: blend
steps_layout: flat
authors: ntn-x2, kashepavadan
owner: logos
doc_version: 1
slug: join-the-blend-network-as-a-core-node
---

# Join the Blend Network as a core node

#### Connect your blockchain node to Blend to contribute to proposer privacy.

Joining the Blend Network lets your blockchain node contribute to the privacy of Logos Blockchain proposers and receive rewards for participating. This procedure applies to operators of a running Logos Blockchain node who want to register that node as a Blend core node. Before you start, make sure your node's address is publicly reachable so other peers can connect to it.

You need:

> - [A running blockchain node](../get-started/run-a-logos-blockchain-node-from-cli.md).

Reviewer: The prerequisite link installs the public-testnet release `0.1.2`, which doesn't expose the Blend API. Joining as a core node needs a `0.1.3-rc.*` devnet build — `/blend/join` and `/mantle/sdp/declarations` exist there, not in `0.1.2`. Worth stating in the prerequisites:

```suggestion
- [A running blockchain node](../get-started/run-a-logos-blockchain-node-from-cli.md) on a `0.1.3-rc.*` devnet build. The Blend API used below (`POST /blend/join`, `GET /mantle/sdp/declarations`) is not in the public-testnet `0.1.2` release.
```

Reviewer: I would duplicate this link to the `1. Start the node and wait for it to switch to "Online" mode. This takes approximately one hour.` as some people just skip to the lists and miss prerequisites etc

- A publicly reachable IP and port (or DNS) combination for the node.

## What to expect

- Your blockchain node is registered as a Blend core node.
- Your node contributes to proposer privacy and becomes eligible for rewards.
- Your declaration is confirmed on-chain and becomes active after two epochs.

## Register your node as a Blend core node

Complete these steps to fund the required keys, retrieve a locked note, and submit your Blend join declaration.

> 1. Start the node and wait for it to switch to "Online" mode. This takes approximately one hour.

Reviewer: **Sync time conflicts with the prerequisite guide.**

The prerequisite doc ([Run a Logos Blockchain node from CLI](../get-started/run-a-logos-blockchain-node-from-cli.md)) says _"Bootstrapping may take 12 to 24 hours."_ — not one hour.

```suggestion
1. Start the node and wait for it to switch to "Online" mode. Bootstrapping may take 12 to 24 hours.
```

Author: This is old, I've updated the blockchain node tutorial accordingly

1. Use the [faucet](https://testnet.blockchain.logos.co/web/faucet/) to send funds to the public key specified in the `user_config.yaml` file, in `sdp.wallet.funding_pk`.
1. Use the faucet to send funds to the public key specified in the `user_config.yaml` file, in `blend.core.zk.secret_key_kms_id`.

> 1. Call the `<YOUR_NODE_IP>/wallet/<pk>/balance` endpoint, using the key from the previous step as `<pk>`, to view the balances of your notes.

Reviewer: The API URLs need a scheme and port — the node listens on `:8080`. Same fix applies to the `/blend/join` URL (line 39) and `/mantle/sdp/declarations` (line 48).

```suggestion
1. Call the `http://<YOUR_NODE_IP>:8080/wallet/<pk>/balance` endpoint, using the value from `blend.core.zk.secret_key_kms_id` as `<pk>`, to view the balances of your notes.
```

> 1. From the response, select either one of the returned notes.

Reviewer: Step 5 says "select one of the returned notes" and step 6 needs a `locked_note_id`, but step 4 only fetches a balance — the doc never shows where the note ID comes from. Please show the response field (or endpoint) that gives `locked_note_id`.

1. Send a `POST` request to `<YOUR_NODE_IP>/blend/join` with the following payload:

   ```bash

>    {"locator": <multiaddr>, "locked_note_id": <node_id>}

Reviewer: `locked_note_id` must be a quoted 64-hex string (the API rejects unquoted and non-string values), and the placeholder should match the `<locked_note_id>` described below.

```suggestion
   {"locator": "<multiaddr>", "locked_note_id": "<locked_note_id>"}
```
